### ISSUE TITLE
aws-pipes-alpha: `InputTransformation` in `Pipe` should automatically add expression delimiters

### ISSUE THREAD

**Issue by garysassano** [aws/aws-cdk#32416]
### Describe the bug

When using the AWS Pipes L2 construct, there's an inconsistency in how JSON paths are handled between `queryStringParameters` and `inputTransformation`.

For `queryStringParameters`, JSON paths work directly:

```typescript
queryStringParameters: {
  key: "$.dynamodb.Keys.Location.S",
  ttl_seconds: "$.dynamodb.NewImage.TTL.N"
}
```

However, for `inputTransformation`, we need to manually add expression delimiters (`<>`) around JSON paths:

```typescript
inputTransformation: InputTransformation.fromObject({
  Location: "<$.dynamodb.Keys.Location.S>",
  MaxTemp: "<$.dynamodb.NewImage.MaxTemp.N>",
  MinTemp: "<$.dynamodb.NewImage.MinTemp.N>",
  ChancesOfPrecipitation: "<$.dynamodb.NewImage.ChancesOfPrecipitation.N>"
})
```

This creates an inconsistent developer experience and increases the chance of errors if developers forget to add the delimiters.

### Suggested Solution
The L2 construct should automatically add the required expression delimiters (`<>`) around JSON paths in `inputTransformation` objects, similar to how `queryStringParameters` handles paths directly.

Example of desired API:

```typescript
inputTransformation: InputTransformation.fromObject({
  Location: "$.dynamodb.Keys.Location.S",
  MaxTemp: "$.dynamodb.NewImage.MaxTemp.N",
  MinTemp: "$.dynamodb.NewImage.MinTemp.N",
  ChancesOfPrecipitation: "$.dynamodb.NewImage.ChancesOfPrecipitation.N"
})
```

The L2 construct could detect JSON paths (starting with `$.`) and automatically wrap them in expression delimiters, improving developer experience and preventing errors from forgotten delimiters.

### Regression Issue

- [ ] Select this option if this issue appears to be a regression.

### Expected Behavior

See above.

### Current Behavior

See above.

### Reproduction Steps

See above.

### CDK CLI Version

2.172.0

### Node.js Version

22.11.0

### OS

Ubuntu 24.04

### Language

TypeScript

**Comment by pahud** [comment]
Yes this seems to be inconsistent developer experience. Not sure if this is intentional. We'll bring this up to the team for inputs.

**Comment by aaythapa** [comment]
`queryStringParameters` is passed straight through to the CFN resource. Presumably this is how CFN expects the queries to be formatted, we don't handle anything there. 

As for `inputTransformation` it takes in the `InputTransformation` object which by default you have to add expression delimiters (<>) around it but there is an `DynamicInput` object defined [here](https://github.com/aws/aws-cdk/blob/4e71675f2a164644eb9aefbb6378c590647cd0a8/packages/%40aws-cdk/aws-pipes-alpha/lib/pipeVariable.ts#L45-L54) which has a `fromEventPath` method that can add the delimiters for you. So the code from the issue description looks like this 

```
inputTransformation: InputTransformation.fromObject({
  Location: DynamicInput.fromEventPath("$.dynamodb.Keys.Location.S"),
  MaxTemp: DynamicInput.fromEventPath("$.dynamodb.NewImage.MaxTemp.N"),
  MinTemp: DynamicInput.fromEventPath("$.dynamodb.NewImage.MinTemp.N"),
  ChancesOfPrecipitation: DynamicInput.fromEventPath("$.dynamodb.NewImage.ChancesOfPrecipitation.N")
})
```
example of it being used in an integ test [here](https://github.com/aws/aws-cdk/blob/4e71675f2a164644eb9aefbb6378c590647cd0a8/packages/%40aws-cdk/aws-pipes-alpha/test/integ.pipe.ts#L66-L69). 

I'm not too sure why by default we don't add `<>` and I couldn't find any comment addressing that in the RFC for this module [here](https://github.com/aws/aws-cdk-rfcs/pull/488). I suspect its so that we only change the input if customers use `DynamicInput`, otherwise we leave it as is and don't add anything behind the scenes by default.

Since this issue has already been addressed I'm going to close this. Feel free to re-open if you disagree
